### PR TITLE
adding parsnp 2.0.4

### PR DESCRIPTION
**Pull request by erinyoung**
There's a new version of parsnp!!!

Well, actually a few versions.

https://github.com/marbl/parsnp/releases

The prior Dockerfile was... old. I made a lot of changes to it, but also kept a lot of lines from the builder stage. I also didn't really change the README associated with it other than add the tool versions.

I have made many changes:
- updated labels
- updated base image to ubuntu:jammy
- added fastani since the documentation recommended it
- added parsnp to both the main readme and the program license readme (I guess it was forgotten... oops!)
- moved the tests from a separate script to the main Dockerfile
- added a CMD line

I can make more changes if need be.

Pull Request (PR) checklist:
- [X] Include a description of what is in this pull request in this message.
- [X] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [X] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [X] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [X] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [X] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [X] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [X] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [X] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

File: Program_Licenses.md [hunk 1]
```diff
@@ -109,6 +109,7 @@ The licenses of the open-source software that is contained in these Docker image
 | OrthoFinder | GNU GPLv3 | https://github.com/davidemms/OrthoFinder/blob/master/License.md |
 | Panaroo | MIT | https://github.com/gtonkinhill/panaroo/blob/master/LICENSE |
 | Pangolin | GNU GPLv3 | https://github.com/cov-lineages/pangolin/blob/master/LICENSE.txt |
+| Parsnp | Battelle National Biodefense Institute (BNBI) | https://github.com/marbl/parsnp?tab=License-1-ov-file#readme |
```
Comment (Kincekara) [inline Program_Licenses.md:112]: In case you missed it, https://github.com/marbl/parsnp/blob/master/LICENSE

**Comment by Kincekara** [comment]
@erinyoung It has a little tricky license. I am not exactly sure, but we may need to add the license file to the image too.  If you say there is no need, I will merge it as is.

>2. Redistributions in binary form must reproduce the above copyright
notice, this list of conditions and the following disclaimer in the
documentation and/or other materials provided with the distribution.

**Comment by erinyoung** [comment]
@Kincekara , good catch! 

I've added a line that copies the license into the container. Do you think `/LICENSE` is a good place?

**Comment by Kincekara** [comment]
@erinyoung, I just realized that source files are copied to the app stage. LICENSE is already in there. 
I am not familiar with parsnp. Is there anything necessary in there after building? Depending on your answer either 
`COPY --from=builder /parsnp-$PARSNP_VER/ /parsnp/` or `COPY --from=builder /parsnp-$PARSNP_VER/LICENSE /.` is redundant. I am sorry for my confusion.

**Comment by Kincekara** [comment]
Okay! I am merging and deploying this PR.

**Review by Kincekara — APPROVED** [review]

**Comment by Kincekara** [comment]
Thank you for putting all this together!
You can check the image deployment from here: https://github.com/StaPH-B/docker-builds/actions/runs/8331854107.

**Pull request merged**